### DISCUSSION TITLE
Bluetooth Gamepads Not Found In Linux

### DISCUSSION THREAD

**Discussion by ezacharyk** [HaxeFlixel/flixel#2421] [Q&A]
Ok. I have been working on and testing controllers like mad for the last week. And I have found and issue with Bluetooth gamepads on Linux targets. My Bluetooth Switch Pro Controller and my Ouya controllers could connect to my PC, but were not usable by any game. Steam didn't see them, and my Flixel games couldn't see them.

After debugging this on a device level, I realized that Ubuntu 20.04 had buggy bluetooth drivers built into the kernel it used. So I upgraded to 21.04. Now, my controllers are usable by Steam, but Flixel still does not see them.

I tested this using the GamePad Test demo as well as the FlxAction method in the FlxAction and Mode demos. Only my wired controllers can be seen by Flixel.

I booted up my Windows partition and tested there, and both my Switch Pro and Ouya controllers were usable in both GamePad Test and FlxAction. So the issue lies in the Linux target.

Also, the HTML5 target could see my Ouya controller but not my Switch Pro controller in Windows and Linux.

Where would be a good place to start looking for a fix for this?

**Comment by Geokureli** [comment]
I would send this up the wire to [OpenFL](https://github.com/openfl/openfl) or [Lime](https://github.com/haxelime/lime). I think they have an outdated mappings library, I think they're using [SDL_GameControllerDB](https://github.com/gabomdq/SDL_GameControllerDB). Before posting there, I would try out their [Gamepad Demo](https://github.com/openfl/openfl-samples/tree/master/features/ui/GamepadInput) just to make sure it's not an issue with flixel

It's worth noting that my Switch Pro Controller works fine on my Mac via bluetooth

**Reply by ezacharyk** [reply]
Thanks. I compiled that OpenFl sample and it only showed my wired controllers. So I will follow up there.
